docs/node/run-your-node/keymanager-node/key-manager-upgrade.md

```diff
@@ -0,0 +1,103 @@
+# Upgrading Key Managers
+
+This guide will describe how to upgrade a Key Manager node.
+
+## About the Upgrade
+
+Every Key Manager node contains all the keys used by confidential ParaTimes 
+inside its TEE-encrypted state. The key material is sealed and can only be 
+decrypted by the exactly same TEE enclave running on the exactly same CPU. This 
+means that newer Key Manager paratimes can not read the key material and that 
+key material can not be restored on another machine. 
+
+**During a Key Manager node upgrade it is therefore essential that the key 
+material is not lost, not even due to an operational error or even a 
+catastrophically failed upgrade.**
+
+## Safe Upgrade Procedure
+
+A Key Manager node's upgrade procedure differs from other Oasis nodes upgrades 
+because the upgraded node cannot unseal/decrypt the old Key Manager's state.
+
+To upgrade a Key Manager node, we need to delete the local state and let the 
+Key Manager's state replicate itself from other nodes. Only one Key Manager 
+runtime in the configuration file can be present at once. 
+
+**Always follow the safe upgrade procedure:**
+1. Keep the first half of nodes running on the old version
```

Comment:
Which half? The guide is about upgrading a single node, so this seems a bit confusing. Should it start out as assuming there are two nodes or 2N nodes to upgrade?

Reply:
Changed it a little bit. Hope now is better. I just wanted to say that upgrading all Key Manager nodes at the same time is not a good idea.